modules/standard/IO.chpl

```diff
@@ -5390,7 +5437,12 @@ proc _setIfPrimitive(ref lhs:?t, rhs:?t2, argi:int):syserr where t!=bool&&_isIoP
         lhs = rhs:int:t;
       }
     } else {
-      lhs = rhs:t;
+      if isBytesType(t2) && isStringType(t) {
+        lhs = rhs.decode(DecodePolicy.Ignore);
```

Comment:
My thinking was what happens here pertains to UTF-8 validation, and since it is not implemented yet (as far as I understand), we could simply ignore non-UTF8 characters for now. But I will change it to Strict and see if any issue comes up

Reply:
I thought that "Ignore" meant "drop these characters" rather than "put them in string unmodified"

Reply:
Oops, I got confused. It makes me think, though. May be we need that policy, too. It'd simply amount to copying the buffer entirely without any check. But maybe that shouldn't happen in `decode`.

For this particular case, I agree that `Strict` is more appropriate than `Ignore`